packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2311,7 +2311,7 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
       if (animationController.status != AnimationStatus.dismissed) {
         _dismissedBottomSheets.add(bottomSheet);
       }
-      completer.complete();
+      completer.complete(null as T);
```

Comment:
Wait a minute, is this a safe cast to make?

cc @goderbauer

Reply:
This doesn't look right. Is it just hiding a typing error elsewhere?

Reply:
Please have a look at this particular https://github.com/dart-lang/sdk/issues/51367#issuecomment-1433561886 and https://github.com/dart-lang/sdk/issues/51367#issuecomment-1430392927

The current Flutter usage is actually `null`.

Reply:
My understanding from reading those comments is that the old code is unsafe (and that there may soon be a warning about this). This change silences that warning, but the code still remains unsafe: There is no guarantee that `T` is actually nullable, so this may still throw at runtime. 

So, instead of just making sure the warning doesn't show up, we should actually fix the unsafe code here.